subprojects/platform-jvm/src/integTest/groovy/org/gradle/jvm/toolchain/SharedJavaInstallationRegistryIntegrationTest.groovy

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.AvailableJavaHomes
+
+class SharedJavaInstallationRegistryIntegrationTest extends AbstractIntegrationSpec {
+
+    def "plugin can query information about the current JVM"() {
```

Comment:
This test is about querying at task execution time. It would be good to also cover configuration time queries. Among other things this would help uncover issues with the configuration cache.

Reply:
Thanks for the feedback Paul. Maybe I'm missing something: the line that resolves the installation is `println` which happens at configuration time.

Reply:
Ah, crossed eyes. It seems I dreamt of a `doLast {}` around that `println()` :) 👍 

So I would expect that this test will fail with the configuration cache enabled because of the undeclared read of system properties, see https://docs.gradle.org/nightly/userguide/configuration_cache.html#undeclared_sys_prop_read

Reply:
Interesting. Let's see what the build server says but running with `instantIntegTest` passes locally (shouldn't this be `configurationCacheIntegTest` by now?)

Reply:
Right, we don't instrument core code for sys props read. Tried locally.

A good way to exercise it is to run the build twice in the test with different system properties (e.g. `/unknown/path` and `/another/path`).